### PR TITLE
docs: correct TypeScript version ranges in pre-v9 version compatibility table

### LINKED ISSUE #51643: [docs] Version compatibility guide: TypeScript versions listed for Angular 8.2 + Angular CLI 8.2/8.3 are incorrect
### Describe the problem that you experienced

TypeScript support is unclear for Angular 8.2.x.

#31615 included TypeScript 3.5 support in Angular 8.2.0. Looking at Angular changelogs, it is unclear if TypeScript 3.4 support was dropped in Angular 8.2.0. However, it is clear that both TypeScript 3.4 and 3.5 support was dropped in Angular 9.0.0.

The *Version compatibility* guide is missing an upper version range for the Angular 8.2.x + Angular CLI 8.2.x entry.

### Enter the URL of the topic with the problem

https://angular.io/guide/versions#before-v9

### Describe what you were looking for in the documentation

I was looking for the supported versions of TypeScript in an Angular 8.x application.

### Describe the actions that led you to experience the problem

- The Angular 8.2.x + Angular CLI 8.2.x entry lists TypeScript support as `>=3.4.2` (notice no upper limit)
- The Angular 8.2.x + Angular CLI 8.3.x entry lists TypeScript support as `>= 3.5.3 <3.6.0` - is TypeScript 3.4.x not supported by this combination?

### Describe what you want to experience that would fix the problem

- An upper limit on the Angular 8.2.x + Angular CLI 8.2.x entry's TypeScript version range
- Clarification on whether TypeScript 3.5 support was dropped in Angular 8.2.x and Angular CLI 8.2.x or 8.3.x

### Add a screenshot if that helps illustrate the problem

Angular 8.x version compatibility matrix

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

_No response_

### Provide any additional information here in as much as detail as you can

My original work in https://gist.github.com/LayZeeDK/c822cc812f75bb07b7c55d07ba2719b3 suggests that Angular CLI 8.3.x only supports TypeScript 3.5.x. This was likely based on the defaults in a freshly generated Angular CLI 8.3.x workspace.

I am willing to submit an Angular.io PR when details on version ranges are clarified.

### PR DESCRIPTION
**Pull request by LayZeeDK**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
In the *Version compatibility* document, pre-v9 version ranges for TypeScript are incorrect.

Issue Number: Fixes #51643

## What is the new behavior?
In the *Version compatibility* document, pre-v9 version ranges for TypeScript are corrected.

For versions 5.x through 8.x, this is based on the `package.json` file of `@angular/compiler-cli`.

For versions 2.x through 4.x, this is based on Angular's changelog.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Before change

*Before change*

After change

*After change*

### CODE REVIEW

**Comment by LayZeeDK** [comment]
`CI / framework-win` and `CI / aio-local` job failures seem unrelated. Correct me if I'm wrong.

File: aio/content/guide/versions.md [hunk 1]
```diff
@@ -40,29 +40,21 @@ guarantees. It is listed here for historical reference.
...
-| >=4.0.0 <4.2.0              | 1.1.x              | ^6.9.0              | ~2.3.0             | ^5.0.1             |
-| >=4.0.0 <4.2.0              | 1.0.x              | ^6.9.0              | ~2.2.0             | ^5.0.1             |
-| 2.x                         | -                  | ^6.9.0              | ~2.0.0             | ^5.0.1             |
+### Before v9
+
+Until Angular v9, Angular and Angular CLI versions were not synced.
+
+| Angular                     | Angular CLI                 | Node.js             | TypeScript     | RxJS   |
+| --------------------------- | --------------------------- | ------------------- | -------------- | ------ |
+| 8.2.x                       | 8.2.x \|\|8.3.x             | ^10.9.0             | >=3.4.2 <3.6.0 | ^6.4.0 |
```
Comment (JeanMeche) [inline aio/content/guide/versions.md:49]: A small nit, it looks off without the space ! 
```suggestion
| 8.2.x                       | 8.2.x \|\| 8.3.x             | ^10.9.0             | >=3.4.2 <3.6.0 | ^6.4.0 |
```

**Review by JeanMeche — APPROVED** [review]
A single NIT, else LGTM

**Comment by LayZeeDK** [comment]
Thank you, @JeanMeche

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 26af16c59368dcbc953f34ed5ba93cc63035a93e.

**Comment by AndrewKushnir** [comment]
@LayZeeDK thanks for creating this PR. We've merged it into the main branch, but it had conflicts with 16.2.x branch. Could you please create a version of this PR for the 16.2.x branch? Thank you.

**Pull request closed without merging**